### DISCUSSION TITLE
-partial-alignment option affects the fraction of 'no hits' reads

### DISCUSSION THREAD

**Discussion by EvgeniiaAlekseeva** [milaboratory/mixcr#1026] [Q&A] [answered]
Dear MiXCR,

we reconstruct immune repertoires from FFPE preserved samples using mixcr 3.0.13, characterized by the high fraction of degraded RNA.  We tested partial-alignment option to include more reads in clonotype reconstruction and found a confusing thing. After the alignment of the same set of samples with and without partial-alignment option via following command:

mixcr align -p default \
  --species hsa \
  -OvParameters.geneFeatureToAlign=VTranscript \
  -OvjAlignmentOrder=JThenV \
  -OjParameters.parameters.floatingRightBound=floating \
  -OallowPartialAlignments=true \
  -OsaveOriginalReads=false \
  -t 8 -f -r align.report.txt \
  1.fastq.gz 2.fastq.gz alignments.vdjca

we got very different fractions of reads with 'no TCR/IG hits', however the sum clone count of resulting repertoires is not much affected. What can be the reason for it? Alignment info is on barplots below:

![image](https://user-images.githubusercontent.com/47817278/223415686-e7c06518-6dac-4578-9ec7-c600ab8ab0b6.png)

Best regards,
Evgeniia Alekseeva

**Comment by PoslavskySV** [comment, marked as answer]
Hi Evgeniia,

By default, for amplicon protocols, MiXCR will keep only those alignments at the [`align`](https://mixcr.com/mixcr/reference/mixcr-align/) step, which fully cover CDR3 region. For RNA-Seq protocols, there is a tiny fraction of the alignments covering CDR3 region. Thus, when you run amplicon pipeline on RNA-Seq data, by default you will see a tiny fraction of successfully aligned reads. The option `allowPartialAlignments` instructs MiXCR to keep alignments which do not fully cover CDR3 region. It is required for the partial assembly procedure, which tries to rescue some productive alignments covering only parts of CDR3 region. That's why you see the difference.

Processing of RNA-Seq data is very tricky. MiXCR has a dedicated [pipeline](https://mixcr.com/mixcr/reference/overview-built-in-presets/#rna-seq-data) for this data; you can check details [in this publication]( https://www.nature.com/articles/nbt.3979#Sec1). The command you use for `align` is not optiomal for RNA-Seq data, as it is uses amplicon settings; with MiXCR 3.0.13 it is better to use `mixcr analyze shotgun ...` pipeline which is designed for such data and has a lot of dedicated options (like specifically optimized aligners and partial assembly):

```
mixcr analyze shotgun
    -s <species> \
    --starting-material <startingMaterial> \
    [OPTIONS] input_file1 [input_file2] analysis_name
```

or in a step-by-step way:

```
# align raw reads
mixcr align -s <species> -p rna-seq \
    -OallowPartialAlignments=true\
    -OvParameters.geneFeatureToAlign=<vFeatureToAlign> \ 
    -OvParameters.parameters.floatingLeftBound=false \
    -OvParameters.parameters.floatingRightBound=false \
    -OvParameters.parameters.floatingRightBound=false \
    [align options] input_R1.fastq [input_R2.fastq] my_analysis.vdjca

# assemble overlapping fragmented sequencing reads
mixcr assemblePartial [assemblePartial options] my_analysis.vdjca my_analysis.rescued_1.clna
mixcr assemblePartial [assemblePartial options] my_analysis.rescued_1.vdjca my_analysis.rescued_2.clna

# impute germline sequences for good TCR alignments
mixcr extend [extend options] my_analysis.rescued_2.vdjca my_analysis.rescued_2.extended.vdjca

# assemble CDR3 clonotypes
mixcr assemble --write-alignments [assemble options] my_analysis.rescued_2.extended.vdjca my_analysis.clna

# assemble contigs: execute only if --assembleContigs is specified
mixcr assembleContigs [assembleContigs options] my_analysis.clna my_analysis.clns

# export to tsv
mixcr exportClones [export options] my_analysis.clns my_analysis.txt
```

Best,
Stan

**Reply by EvgeniiaAlekseeva** [reply]
Thank you very much for the reply, it helped!

**Comment by amikelov** [comment]
I can also add, regarding small difference in clone coutns, that clone assembly is a separate step in MiXCR, so more aligned reads does not automatically mean more clones. In this example, the growth in the number of aligned reads is due to reads which does not cover CDR3 (-OallowPartialAlignments means this). A clonal sequence always contains CDR3, so newly added reads won't be able to add any new clonotypes, because clonotypes are defined by CDR3 (at minimum) which is not covered by these additional reads.

**Reply by EvgeniiaAlekseeva** [reply]
Thank you very much for the reply, it helped!
